### PR TITLE
Menu popup dialog Pull Request 

### PR DESCRIPTION
**Pull request by henask12**
# This Pull Request includes the below feature and changes 

- [ ] Incorporated a JavaScript array to efficiently store all project information, providing a more organized structure.

- [ ] Refactored the project section to dynamically generate HTML content using the information stored in the JavaScript object. This ensures that the webpage is created dynamically upon loading, resulting in a smoother user experience.

- [ ] Implemented a captivating feature: when users click or tap the "see project" button, a popup window is displayed
- [ ] implemented a close button for the popup menu
- [ ] and made popup responsive to support desktop and mobile version

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -118,7 +118,7 @@ <h2>
               </li>
             </ul>
             <div class="card-button">
-              <button type="button" class="btn background">
+              <button type="button" class="btn background" data-project-index="0">
                 See Project
```
[the remark this answers is not in the document]
Comment (henask12) [reply index.html:122]: @AbayomiOlaoye 
#Thank you for the kind review

Regarding the second change you mentioned about dynamically generating the works section, I have implemented it using JavaScript. By clicking a button, the page is created and displayed automatically. I have provided a screenshot below for your reference. If you have any further questions or if the screenshot is unclear, please feel free to ask.

image

[the remark this answers is not in the document]
Comment (AbayomiOlaoye) [reply index.html:122]: @henask12 , awesome! In that case, kindly delete the project contents that you have in the `index.html`. You still have those html structures there, which defeats the purpose of the dynamic rendering. Good work, @henask12 💪

[the remark this answers is not in the document]
Comment (AbayomiOlaoye) [reply index.html:122]: Please note that I wasn't referring to the modal pop up. Both the modal and the project cards should be created dynamically. While project cards get create on windows load, pop-up gets created with a click event just like you have done.

[the remark this answers is not in the document]
Comment (henask12) [reply index.html:122]: @AbayomiOlaoye 
yes i think now i understood your point and please check the below if this is the way instead od using `data-project-index` i used ` const projectButtons = Array.from(document.querySelectorAll('.btn'));`
image
image

**Review by divinecharlotte — CHANGES_REQUESTED** [review]
Hi @henask12         , 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## To Highlight 💯 

- linters are passing 👍 
- Descriptive Pr 👍 




 

## Required Changes ♻️
☠️ Next time consider making sure that all the files you touched are in your latest commit to make the review process easier it will also help you to see easily what required changes and where exactly as the reviewer will highlight it

- [ ] Kindly as requested by the previous reviewer Please note that he wasn't referring to the modal pop-up. Both the modal and the project cards should be created dynamically. While project cards get created on Windows load, a pop-up gets created with a click event just like you have done.
- check the image below I highlighted what am saying here
Screen Shot 2023-05-19 at 06 59 24
 

doing so will help you to meet the below requirement 
> The projects section is created dynamically with the information from the projects array (all the HTML of the section is created when the page loads). This requires changes in existing code - already created HTML code for the projects list should be removed, and instead it should be added via JavaScript.  and after the modal popup data is populated with the data from the projects array.

_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @team, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost

## Suggested changes :recycle: 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @henask12 , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Well done implementing the requested changes. Your project is working as expected ✔ 🎉🎉

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html [hunk 2]
```diff
@@ -8,10 +8,11 @@
   <link rel="stylesheet" href="./css/style.css" />
   <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.10.5/font/bootstrap-icons.css">
   <title>Henok Mekonnen | Portfolio</title>
-  <script type="text/javascript" src="./js/cards.js"></script>
+
 </head>
 
 <body>
+  <script type="text/javascript" src="./js/cards.js"></script>
```
[the remark this answers is not in the document]
Comment (henask12) [reply index.html:15]: @ridwanediallo  thank you very much for the suggestion i will read the article

**Pull request merged**